app/main.py

```diff
@@ -1,34 +1,93 @@
+from typing import Union
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int = 9,
+            column: int = 9,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_decks()
+
+    def create_decks(self) -> list[Deck]:
+        decks = list()
+        for row in range(self.start[0], self.end[0] + 1):
+            for column in range(self.start[1], self.end[1] + 1):
+                deck = Deck(row, column)
+                decks.append(deck)
+        return decks
+
+    def get_deck(self, row: int, column: int) -> Union[Deck, None]:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def fire(self, row: int, column: int) -> str:
+        deck = self.get_deck(row, column)
+        if deck:
+            deck.is_alive = False
+            if self.is_ship_drowned():
+                self.is_drowned = True
+                return "Sunk!"
+            return "Hit!"
+        return "Miss!"
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def is_ship_drowned(self) -> bool:
+        for deck in self.decks:
+            if deck.is_alive:
+                return False
+        return True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
+        self.field = self.create_field(ships)
+
+    def create_field(self, ships: list[Ship]) -> dict:
+        field = {}
+        for ship in ships:
+            start, end = ship
```

Comment:
if ships is a list of ships, are you sure, that you can unpack ship object like that into start and end?  It more looks like list[tuple[int, int]]